moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -217,6 +220,139 @@ struct CollisionRequest
   /** \brief Flag indicating whether information about detected collisions should be reported */
   bool verbose;
 };
+
+struct DistanceRequest
+{
+  DistanceRequest()
+    : enable_nearest_points(false)
+    , enable_signed_distance(false)
+    , global(true)
+    , active_components_only(nullptr)
+    , acm(nullptr)
+    , distance_threshold(std::numeric_limits<double>::max())
+    , verbose(false)
+  {
+  }
+
+  virtual ~DistanceRequest()
+  {
+  }
+
+  /// Compute \e active_components_only_ based on \e req_
+  void enableGroup(const robot_model::RobotModelConstPtr& kmodel)
+  {
+    if (kmodel->hasJointModelGroup(group_name))
+      active_components_only = &kmodel->getJointModelGroup(group_name)->getUpdatedLinkModelsWithGeometrySet();
+    else
+      active_components_only = nullptr;
+  }
+
+  /// Indicate if nearest point information should be calculated
+  bool enable_nearest_points;
+
+  /// Indicate if a signed distance should be calculated in a collision.
+  bool enable_signed_distance;
+
+  /// Indicate if the global minimum should only be found. If this is true
+  /// it will only try to find the global minimum distance and not store information
+  /// on a link by link basis. If this is set to false it will store distance information
+  /// for every link in the active_components_only list.
+  bool global;
```

Comment:
`global` is somewhat generic to read. What about `global_minimum_only` or reversing the logic and using `detailed_distances` or `linkwise_distances`?